src/dune_engine/artifact_substitution.ml

```diff
@@ -31,6 +31,7 @@ open Import
 type configpath =
   | Sourceroot
   | Stdlib
+  | OCamlCompiler of [ `Byte | `Native ]
```

Comment:
What about using `Mode.t`?

Also, it should be `Ocaml_compiler` (we call it `Ocaml` everywhere else in the code) or `Ocamlc`.

Reply:
I see. I wouldn't mind leaving the code as is, but I know we don't tolerate polymorphic variants all that much.
Perhaps `Mode`, `Cm_kind`, etc. should be extracted to some `Dune_ocaml` library where they could be accessible?